Automattic/wp-calypso
# Notifications: remove AuthWrapper HOC

Simplifies how the standalone Notifications app creates the `wpcom` client and how it passes it to the root React component.

Until now, there was an `AuthWrapper` HOC and a `useClient` hook, with `useEffect`s and `useState`s to create the `wpcom` client asynchronously, to initialize the analytics library with `setTracksUser` and to render the React UI only after the client is created.

This PR moves that init code outside React, and the boot code is now something like:
```js
createClient().then( wpcom => {
  setTracksUser( wpcom );
  ReactDOM.render( <Notifications wpcom={ wpcom } />, rootEl );
} );
```

I was inspired by @sirbrillig's #52890 that copies the code to the Shopping Cart app.

**How to test:**
Build and start the Notifications app by `cd apps/notifications; yarn dev-server`. Verify that the app at `calypso.localhost:3000` works correctly and that it can connect to the REST API.
